libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/StreamingExtensions/BotFrameworkStreamingExtensionsAdapter.cs

```diff
@@ -0,0 +1,205 @@
+        {
+            if (turnContext == null)
+            {
+                throw new ArgumentNullException(nameof(turnContext));
+            }
+
+            if (activities == null)
+            {
+                throw new ArgumentNullException(nameof(activities));
+            }
+
+            if (activities.Length == 0)
+            {
+                throw new ArgumentException("Expecting one or more activities, but the array was empty.", nameof(activities));
+            }
+
+            var responses = new ResourceResponse[activities.Length];
+
+            /*
+             * NOTE: we're using for here (vs. foreach) because we want to simultaneously index into the
+             * activities array to get the activity to process as well as use that index to assign
+             * the response to the responses array and this is the most cost effective way to do that.
+             */
+            for (var index = 0; index < activities.Length; index++)
+            {
+                var activity = activities[index];
+                var response = default(ResourceResponse);
+                _logger.LogInformation($"Sending activity.  ReplyToId: {activity.ReplyToId}");
+
+                if (activity.Type == ActivityTypesEx.Delay)
+                {
+                    // The Activity Schema doesn't have a delay type build in, so it's simulated
+                    // here in the Bot. This matches the behavior in the Node connector.
+                    var delayMs = (int)activity.Value;
+                    await Task.Delay(delayMs, cancellationToken).ConfigureAwait(false);
+
+                    // No need to create a response. One will be created below.
+                }
+                else if (activity.Type == ActivityTypesEx.InvokeResponse)
+                {
+                    turnContext.TurnState.Add(InvokeReponseKey, activity);
+
+                    // No need to create a response. One will be created below.
+                }
+                else if (activity.Type == ActivityTypes.Trace && activity.ChannelId != "emulator")
+                {
+                    // if it is a Trace activity we only send to the channel if it's the emulator.
+                }
+
+                var baseUrl = activity.ServiceUrl + (activity.ServiceUrl.EndsWith("/") ? string.Empty : "/");
+                var requestPath = $"{baseUrl}v3/conversations/{activity.Conversation.Id}/activities/{activity.Id}";
+                var requestBody = JsonConvert.SerializeObject(activity, SerializationSettings.BotSchemaSerializationSettings);
+                server = WebSocketServerRegistry.GetServerByID(serverID);
+
+                if (server != null)
+                {
+                    var serverResponse = await server.SendAsync(Request.CreateRequest(Request.POST, requestPath, new StringContent(requestBody, System.Text.Encoding.UTF8))).ConfigureAwait(false);
```

Comment:
There are some helpers on Request to do things like:
var r = Request.CreatePost(path);
r.SetBody(whatever)